sdk/core/azure-core/test/ut/global_context.cpp

```diff
@@ -24,12 +25,12 @@ TEST(Context, ApplicationContext)
 {
   Context appContext = Context::GetApplicationContext();
 
-  EXPECT_FALSE(appContext.HasKey("Key"));
-  EXPECT_FALSE(appContext.HasKey("key"));
-  EXPECT_FALSE(appContext.HasKey("Value"));
-  EXPECT_FALSE(appContext.HasKey("value"));
-  EXPECT_FALSE(appContext.HasKey("1"));
-  EXPECT_FALSE(appContext.HasKey(""));
+  EXPECT_THROW(appContext.Get<int>("Key"), std::logic_error);
```

Comment:
@JeffreyRichter should `context.GetValue<...>()` throw if the key is not found or should it abort?

Reply:
Also, do we expect the caller to know the type of the value corresponding to the key, to pass in as the template argument ahead of time?

Previously, calling `HasKey` didn't require knowledge of the `T`, but now the user needs to know and pass the right type to the `TryGetValue` method.